scripts/backup.sh

```
@@ -0,0 +1,282 @@
+# numfmt fallback function for systems that don't have it
+format_bytes() {
+    local bytes=$1
+    local size_names=("B" "KB" "MB" "GB" "TB")
+    local i=0
+    local result
+    
+    if command -v numfmt >/dev/null 2>&1; then
+        numfmt --to=iec-i --suffix=B --format=%.1f "$bytes"
+    else
+        # Simple fallback implementation
+        result=$(echo "$bytes")
```

Shellcheck (0.10.0)

[style] Useless echo? Instead of 'cmd $(echo foo)', just use 'cmd foo'.

(SC2116)